spdx-utils/src/main/kotlin/Extensions.kt

```diff
@@ -26,6 +26,37 @@ import java.util.EnumSet
  */
 operator fun <E : Enum<E>> EnumSet<E>.plus(other: EnumSet<E>): EnumSet<E> = EnumSet.copyOf(this).apply { addAll(other) }
 
+/**
+ * Check that the map to be created from the list of pairs would not contain any identical keys. Throws an
+ * [IllegalArgumentException] if such keys are found.
+ */
+fun <V> List<Pair<String, V>>.checkForSameKeyCaptialization(): Map<String, V> =
+    toMap().also {
+        val keys = unzip().first.toMutableList()
+        val uniqueKeys = keys.distinct()
+        if (keys.size > uniqueKeys.size) {
+            uniqueKeys.forEach { uniqueKey -> keys.remove(uniqueKey) }
+            require(keys.isEmpty()) {
+                val quotedKeys = keys.map { "\"$it\"" }
+                "The following ${keys.size} keys are present in the same capitalization: $quotedKeys"
+            }
+        }
+    }
+
+/**
+ * Check that the map does not contain any keys in different capitalizations. Throws an [IllegalArgumentException] if
+ * such keys are found.
+ */
+fun <V> Map<String, V>.checkForDifferentKeyCaptialization(): Map<String, V> =
```

Comment:
This also converts the map to be case-insensitive, I would suggest to move this out of the function and call it after the check. This would be a bit slower because the keys would have to be compared case-insensitively twice (maybe using `distinctBy { it.toLowerCase() }` in the check), but this is not big problem because it is still fast and only executed once.

Apart from that I would prefer if those checks would be unit tests instead of being executed at runtime.

Reply:
> Apart from that I would prefer if those checks would be unit tests instead of being executed at runtime.

That's a good idea basically, and I believe we had discussed it before, but the "problem" with that approach is that we would need to keep the list of pairs around in addition to the map, because you can only check the list of pairs for duplicate keys.

Would that be fine for you?

Reply:
Yes.

Reply:
Done.